apps/src/music/views/GenerateCode.tsx

```diff
@@ -85,6 +78,34 @@ const GenerateCode: React.FunctionComponent<GenerateCodeProps> = ({
       ? adlibs[adlibOption]
       : undefined);
 
+  const getInitialChoices = () => {
+    if (!useAdlib) return {};
+    const initial: AdlibChoices = {};
+    Object.keys(useAdlib.options).forEach(key => {
+      const opts = useAdlib.options[key];
+      initial[key] = sample(opts)?.id || '';
+    });
+    return initial;
+  };
+
+  const [adlibChoices, setAdlibChoices] = useState<AdlibChoices>(
+    getInitialChoices()
+  );
+
+  const [contextText, setContextText] = useState(DefaultContext);
+
+  const [promptText, setPromptText] = useState(
+    adlibOption ? '' : DefaultPrompt
+  );
+
+  useEffect(() => {
+    // If we are clearing, make sure we are called with the new
+    // block count before deciding what to do next.
+    if (aiGenerateState === 'clearing' && blockCount <= 1) {
+      dispatch(setAiGenerateState('none'));
+    }
+  }, [aiGenerateState, blockCount, dispatch]);
```

Comment:
This works around a tricky situation in which the component was being re-rendered immediately after the workspace was cleared but with the previous block count.

Might revisit at some point, but in the meantime, this shows the risk associated with changing any of the flows this close to release.

Reply:
Good catch, this is tricky indeed! An idea for later is, instead of tracking workspace block count, we should pass in a flag of if the level has been modified from starter code (initialSources !== startSources). This value will _not_ change as we modify blocks, since it only applies to the code that was initially loaded when loading the level.